### PR TITLE
[3608] Improve site-component datasource

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/3608

### CODE REVIEW

File: ui/app/src/services/content.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { get } from "../utils/ajax";
+import { map } from 'rxjs/operators';
+import { Observable } from "rxjs";
+
+export function getContent(site: string, path: string): Observable<string> {
```
Comment (rart) [inline ui/app/src/services/content.ts:22]: This one should directly return the string, not the object with the content property

File: static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              });
+              _self.list = items;
+
+              for(var j=0; j<_self.callbacks.length; j++) {
+                _self.callbacks[j].success(items);
+              }
+            },
+            () => {
+              CStudioAuthoring.Operations.showSimpleDialog(
+                "unableLoad-dialog",
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js]: Show new error state dialog

File: static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    for (var i = 0; i < properties.length; i++) {
+      var property = properties[i];
+      if (property.name === "componentPath") {
+        CrafterCMSNext.services.content.getDOM(CStudioAuthoringContext.siteId, property.value)
+          .subscribe(
+            (dom) => {
+              let items = Array.from(dom.querySelectorAll('items > item'));
+              items = items.map((item) => {
+                return { key: item.querySelector('key').innerHTML, value: item.querySelector('value').innerHTML }
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js]: Format, semi colon

File: static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  };
+
+YAHOO.extend(CStudioForms.Datasources.TaxonomySelector, CStudioForms.CStudioFormDatasource, {
+
+  getLabel: function () {
+    return CrafterCMSNext.i18n.intl.formatMessage(this.messages.taxonomySelectorMessages.label);
+  },
+
+  getInterface: function () {
+    return "item";
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js]: reformat whole doc with single quotes

File: static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    }, {
+      label: CrafterCMSNext.i18n.intl.formatMessage(this.messages.taxonomySelectorMessages.componentPath),
+      name: "componentPath",
+      type: "string"
+    }];
+  },
+
+  getSupportedConstraints: function () {
+    return [
+      {label: CrafterCMSNext.i18n.intl.formatMessage(this.messages.taxonomySelectorMessages.required), name: "required", type: "boolean"}
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js]: format, semi

**Review by rart — APPROVED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
Approved pressed by mistake.

See individual comments.

File: static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js (hunk 2 above)
Comment (aachavarria) [reply static-assets/components/cstudio-forms/data-sources/taxonomy-selector.js]: we need to check this tomorrow

**Review by rart — APPROVED** [review]

**Pull request merged**